.github/workflows/build-old-docs.yml

```diff
@@ -0,0 +1,126 @@
+name: Build old docs
+
+on:
+  workflow_dispatch:
+    inputs:
+      release_token:
+        description: 'Your release token'
+        required: true
+      triggered_by:
+        description: 'CD | TAG | MANUAL'
+        required: false
+        default: MANUAL
+      package:
+        description: The name of the repo to build documentation for.
+        type: string
+        default: docarray
+      repo_owner:
+        description: The owner of the repo to build documentation for. Defaults to 'jina-ai'.
+        type: string
+        default: jina-ai
+      pages_branch:
+        description: Branch that Github Pages observes
+        type: string
+        default: gh-pages
+      git_config_name:
+        type: string
+        default: Jina Dev Bot
+      git_config_email:
+        type: string
+        default: dev-bot@jina.ai
+
+jobs:
+  token-check:
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "Success!"
+        if: "${{ github.event.inputs.release_token }} == ${{ env.release_token }}"
+        env:
+          release_token: ${{ secrets.DOCARRAY_RELEASE_TOKEN }}
+      - id: get_versions
+        run: |
+          printf "versions=" >> $GITHUB_OUTPUT
+          curl https://raw.githubusercontent.com/${{ inputs.repo_owner }}/${{ inputs.package }}/main/docs/_versions.json >> $GITHUB_OUTPUT
+    outputs:
+      versions: ${{ steps.get_versions.outputs.versions }}
+
+  build-doc:
+    needs: token-check
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include: ${{ fromJson(needs.token-check.outputs.versions) }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+          ref: ${{ matrix.version }}
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.7'
+      - name: Get latest templates
+        run: |
+          git show --summary
+          echo "Get latest sidebar brand template"
+          wget https://raw.githubusercontent.com/${{ inputs.repo_owner }}/${{ inputs.package }}/main/docs/_templates/sidebar/brand.html
+          mv ./brand.html ./docs/_templates/sidebar/brand.html
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install wheel
+          # pip does not properly resolve dependency versions with syntax pip install --no-cache-dir ".[test,full]"
+          pip install --no-cache-dir ".[test]"
+          pip install --no-cache-dir ".[full]"
+          pip install --no-cache-dir ".[qdrant]"
+          pip install --no-cache-dir ".[annlite]"
+          pip install --no-cache-dir ".[weaviate]"
+          pip install --no-cache-dir ".[elasticsearch]"
+          pip install --no-cache-dir ".[redis]"
+          cd docs
+          pip install -r requirements.txt
+          pip install --pre -U furo
+          pip install sphinx-markdown-tables==0.0.17
```

Comment:
Why isn't this part of requirements.txt?

Reply:
No particular reason other than I forgot 😅 . Thanks for pointing it out

Reply:
Testing the pipeline with the version pin in requirements.txt now

Reply:
Oh wait it doesnt work. Because the requirements.txt for old doc builds will use the old requirements.txt

Reply:
Yea its a bit ugly to pin in post like this. We could go back in time and pin the requirements.txt of the individual releases but might be overkill considering this works atm.